test/e2e/framework/pv_util.go

```diff
@@ -350,85 +421,131 @@ func WaitOnPVandPVC(c clientset.Interface, ns string, pv *v1.PersistentVolume, p
 //   extending test time. Thus, it is recommended to keep the delta between PVs and PVCs
 //   small.
-func WaitAndVerifyBinds(c clientset.Interface, ns string, pvols PVMap, claims PVCMap, testExpected bool) {
+func WaitAndVerifyBinds(c clientset.Interface, ns string, pvols PVMap, claims PVCMap, testExpected bool) error {
 
+	var err error
 	var actualBinds int
 	expectedBinds := len(pvols)
 	if expectedBinds > len(claims) { // want the min of # pvs or #pvcs
 		expectedBinds = len(claims)
 	}
 
 	for pvName := range pvols {
-		err := WaitForPersistentVolumePhase(v1.VolumeBound, c, pvName, 3*time.Second, 180*time.Second)
+		err = WaitForPersistentVolumePhase(v1.VolumeBound, c, pvName, 3*time.Second, 180*time.Second)
 		if err != nil && len(pvols) > len(claims) {
 			Logf("WARN: pv %v is not bound after max wait", pvName)
 			Logf("      This may be ok since there are more pvs than pvcs")
 			continue
 		}
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return fmt.Errorf("PV %q did not become Bound: %v", pvName, err)
+		}
 
 		pv, err := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred())
-		if cr := pv.Spec.ClaimRef; cr != nil && len(cr.Name) > 0 {
+		if err != nil {
+			return fmt.Errorf("PV Get API error: %v", err)
+		}
+		cr := pv.Spec.ClaimRef
+		if cr != nil && len(cr.Name) > 0 {
 			// Assert bound pvc is a test resource. Failing assertion could
 			// indicate non-test PVC interference or a bug in the test
 			pvcKey := makePvcKey(ns, cr.Name)
-			_, found := claims[pvcKey]
-			Expect(found).To(BeTrue(), fmt.Sprintf("PersistentVolume (%q) ClaimRef (%q) does not match any test claims.", pv.Name, cr.Name))
+			if _, found := claims[pvcKey]; !found {
+				return fmt.Errorf("internal: claims map is missing pvc %q", pvcKey)
+			}
 
-			err = WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, cr.Name, 3*time.Second, 180*time.Second)
-			Expect(err).NotTo(HaveOccurred())
+			if err = WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, cr.Name, 3*time.Second, 180*time.Second); err != nil {
+				return fmt.Errorf("PVC %q did not become Bound: %v", cr.Name, err)
+			}
 			actualBinds++
 		}
 	}
 
-	if testExpected {
-		Expect(actualBinds).To(Equal(expectedBinds))
+	if testExpected && actualBinds != expectedBinds {
+		return fmt.Errorf("Expect number of bound PVs (%v) to equal number of claims (%v)", actualBinds, expectedBinds)
 	}
+	return nil
 }
 
 // Test the pod's exit code to be zero.
-func testPodSuccessOrFail(c clientset.Interface, ns string, pod *v1.Pod) {
+func testPodSuccessOrFail(c clientset.Interface, ns string, pod *v1.Pod) error {
 
 	By("Pod should terminate with exitcode 0 (success)")
-	err := WaitForPodSuccessInNamespace(c, pod.Name, ns)
-	Expect(err).NotTo(HaveOccurred())
+	if err := WaitForPodSuccessInNamespace(c, pod.Name, ns); err != nil {
+		fmt.Errorf("pod %q failed to reach Success: %v", pod.Name, err)
```

Comment:
may be you missed to return 
`return fmt.Errorf("pod %q failed to reach Success: %v", pod.Name, err)`